src/components/TextInput/TextInput.tsx

```diff
@@ -1,10 +1,26 @@
 /* eslint-disable react/jsx-props-no-spreading */
 import { Typography } from '@mui/material';
 import InputBase from '@mui/material/InputBase';
+import { UseFormRegister } from 'react-hook-form/dist/types/form';
 
 import styles from './TextInput.module.css';
 
-const TextInput = ({ placeholder, register, error, password, disabled, defaultValue, readOnly }) => {
+interface Error {
```

Comment:
It's a good idea to have a separate interface for an error but the name `Error` is in conflict with a built-in name. Shouldn't be a problem but consider changing the name slightly or changing the interface below so that it has:
```
  error?: {
    message: string;
  };
```